source/common/network/connection_impl.cc

```diff
@@ -111,7 +111,8 @@ void ConnectionImpl::close(ConnectionCloseType type) {
     ASSERT(type == ConnectionCloseType::FlushWrite);
     close_with_flush_ = true;
     read_enabled_ = false;
-    file_event_->setEnabled(Event::FileReadyType::Write | Event::FileReadyType::Closed);
+    file_event_->setEnabled(Event::FileReadyType::Write |
+                            (enable_half_close_ ? 0 : Event::FileReadyType::Closed));
```

Comment:
Ditto here.  I guess you could really just set them all false by default and see what fails :-P